File: app/src/main/java/com/u1fukui/bbs/ui/main/mypage/FavoriteThreadListFragment.java
package com.u1fukui.bbs.ui.main.mypage;

import com.u1fukui.bbs.repository.FavoriteThreadListRepository;
import com.u1fukui.bbs.repository.ThreadListRepository;
import com.u1fukui.bbs.ui.main.BaseThreadListFragment;

public class FavoriteThreadListFragment extends BaseThreadListFragment {

    public static final FavoriteThreadListFragment newInstance() {
        return new FavoriteThreadListFragment();
    }

    public FavoriteThreadListFragment() {
    }

    @Override
    protected ThreadListRepository getRepository() {
        return new FavoriteThreadListRepository();
    }

}

File: app/src/main/java/com/u1fukui/bbs/ui/main/mypage/FavoriteThreadListFragment.kt
package com.u1fukui.bbs.ui.main.mypage

import com.u1fukui.bbs.api.ThreadListApi
import com.u1fukui.bbs.repository.FavoriteThreadListRepository
import com.u1fukui.bbs.ui.main.BaseThreadListFragment
import javax.inject.Inject

class FavoriteThreadListFragment : BaseThreadListFragment() {

    @Inject
    lateinit var threadListApi: ThreadListApi

    override fun getRepository() = FavoriteThreadListRepository(threadListApi)

    companion object {

        fun newInstance() = FavoriteThreadListFragment()
    }

}
